File: Week_01/G20200343040030/move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
  let j = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] != 0) {
      let temp = nums[i];
      nums[i] = 0;
      nums[j] = temp;
      j++;
    }
  }
};

File: Week_01/G20200343040254/MoveZeroes.java
/**
 * 
 */
package week01;

/**
 * @author Jarod.xu
 *
 */
public class MoveZeroes {
	public void moveZeroes(int[] nums) {

		int j = 0;
		for (int i = 0; i < nums.length; i++) {
			if (nums[i] != 0) {
				int temp = nums[j];
				nums[j] = nums[i];
				if (i != j) {
					nums[i] = temp;
				}
				j++;

				// int temp = nums[j];
				// nums[j] = nums[i];
				// nums[i] = temp;
				// j++;
			}
		}

	}
}
